lib/src/Base/Stat/openturns/CovarianceModelImplementation.hxx

```diff
@@ -109,6 +109,11 @@ public:
   virtual CovarianceMatrix discretize(const Sample & vertices) const;
   virtual Sample discretizeRow(const Sample & vertices,
                                const UnsignedInteger p) const;
+  virtual Matrix computeCrossCovariance(const Sample &firstSample,
+                                        const Sample &secondSample) const;
+
+  virtual Matrix computeCrossCovariance(const Sample &sample,
+                                        const Point &point) const;
```

Comment:
I will see. If it takes few time to make changes and have a performant method thus why not.
Otherwise a basic (and not performant) solution is to define things using the `transpose` operator:
```c++
Matrix CovarianceModelImplementation::computeCrossCovariance(const Point &point, const Sample &sample) const
{
  return computeCrossCovariance(sample, point).transpose();
}
```

Note that this kind of method is intensively used (in kriging) with the proposed form `(sample, point)`. I have no clue on potential application with the proposed form but agreed it can disturb users...

Reply:
I see you have implemented the form `(point, sample)`. If it is not often used in practice, then its lack of performance is not an issue. By the same token, couldn't you simply implement the `(point, point)` form as

```c++
Matrix CovarianceModelImplementation::computeCrossCovariance(const Point &point1, const Point &point2) const
{
  return computeCrossCovariance(Sample(1, point1), point2);
}
```
?

Reply:
I implemented a basic one but honestly it is not very efficient. I will think about a better implementation after vacations.
```cxx
computeCrossCovariance(const Point &, const Sample &)
```
is useful in a kriging context, while predicting on a `Point` conditionally to the observations (`Sample`)
The case you mentioned takes no sense for me (at least in a kriging context)